ddmark/validation.go

```diff
@@ -241,3 +297,53 @@ func parseIntOrUInt(value reflect.Value) (int, bool) {
 
 	return fieldInt, ok
 }
+
+// checkValueExistOrIsValid checks if a given string marker item name value exist in a unmarshalled struct (converted to a map by structValueToMap)
+// it returns 1 if the value is found and -if applicable- the required value is valid, 0 otherwise
+func checkValueExistsOrIsValid(markerItem string, structMap map[string]interface{}, ruleName string) (int, error) {
+	// marketItem can either be a fieldName, or fieldName=fieldValue
+	markerItemArr := strings.Split(markerItem, "=")
+
+	switch len(markerItemArr) {
+	// no given value to respect => check item is not null
+	case 1:
+		if structMap[markerItem] != nil {
+			return 1, nil
+		}
+	// a value was required => check item has described value
+	case 2:
+		markerSubfieldName, markerSubfieldValue := markerItemArr[0], markerItemArr[1]
+		if structMap[markerSubfieldName] == nil {
+			break
+		}
+
+		if reflect.Indirect(reflect.ValueOf(structMap[markerSubfieldName])).Type().ConvertibleTo(reflect.TypeOf(markerSubfieldValue)) {
+			v := reflect.Indirect(reflect.ValueOf(structMap[markerSubfieldName]))
+			tv := v.Type()
+			t := reflect.TypeOf(markerSubfieldValue)
+
+			var vStr string
+
+			switch tv.Kind() {
+			case reflect.Int:
+				vInt := v.Convert(tv).Interface().(int)
+				vStr = strconv.Itoa(vInt)
+			case reflect.String:
+				vStr = v.Convert(t).Interface().(string)
+			default:
+				return 0, fmt.Errorf("%v: please do not apply this marker to anything else than int or string. Current type: %v", ruleName, tv.Name())
+			}
+
+			if strings.Compare(markerSubfieldValue, vStr) == 0 {
```

Comment:
```suggestion
			if markerSubfieldValue == vStr {
```

Using string.Compare is only relevant if you use the various output (-1/1/0) here as it's the simple equality it feels more natural to do a natural comparison (case sensitive).

You could add the option to check for equality without case sensitivity (EqualFold)